lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -380,9 +380,9 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
             self.ssh.get_transport().set_keepalive(5)
             chan = self.ssh.get_transport().open_session()
         except Exception as e:
-            msg = "Failed to open session"
-            if len(str(e)) > 0:
-                msg += ": %s" % str(e)
+            msg = u"Failed to open session"
+            if len(to_text(e)) > 0:
+                msg += u": %s" % to_text(e)
             raise AnsibleConnectionFailure(msg)
```

Comment:
@mkrizek @abadger are we allowed to put unicode into our own custom exceptions? Is this handled internally?

Reply:
Only give native strings to exceptions

Reply:
Completely missed this for some reason. Good catch, thanks.